paddle/fluid/operators/mkldnn/interpolate_mkldnn_op.cc

```diff
@@ -0,0 +1,208 @@
+/* Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License. */
+
+#include "paddle/fluid/framework/data_layout_transform.h"
+#include "paddle/fluid/operators/interpolate_op.h"
+#include "paddle/fluid/platform/mkldnn_reuse.h"
+
+namespace paddle {
+namespace operators {
+
+using framework::DataLayout;
+using dnnl::memory;
+using dnnl::primitive;
+using dnnl::reorder;
+using dnnl::stream;
+using dnnl::resampling_forward;
+using platform::GetMKLDNNFormat;
+using platform::to_void_cast;
+
+template <typename T = float>
+class InterpolateMKLDNNHandler
+    : public platform::MKLDNNHandlerT<T, dnnl::resampling_forward> {
+ public:
+  InterpolateMKLDNNHandler(const dnnl::algorithm algo,
+                           const paddle::platform::MKLDNNDeviceContext& dev_ctx,
+                           const dnnl::engine engine, platform::Place cpu_place,
+                           const Tensor* x, Tensor* z,
+                           const std::string& uniq_name)
+      : platform::MKLDNNHandlerT<T, dnnl::resampling_forward>(
+            dev_ctx, engine, cpu_place,
+            platform::CreateKey(
+                framework::vectorize(x->dims()),
+                uniq_name + (algo == dnnl::algorithm::resampling_nearest
+                                 ? "N"
+                                 : "L"))) {
+    if (!this->isCached()) {
+      const auto src_x_tz = framework::vectorize(x->dims());
+      const auto dst_tz = framework::vectorize(z->dims());
+      const auto src0_md = dnnl::memory::desc(
+          src_x_tz, platform::MKLDNNGetDataType<T>(), x->format());
+      const auto dst_md = memory::desc(dst_tz, platform::MKLDNNGetDataType<T>(),
+                                       MKLDNNMemoryFormat::any);
+      auto resampling_d = dnnl::resampling_forward::desc(
+          dnnl::prop_kind::forward_inference, algo, src0_md, dst_md);  // scale
+
+      this->fwd_pd_.reset(new dnnl::resampling_forward::primitive_desc(
+          resampling_d, this->engine_));
+
+      auto key_pd = this->key_ + "@fwd_pd";
+      this->dev_ctx_.SetBlob(key_pd, this->fwd_pd_);
+    }
+  }
+
+  std::shared_ptr<resampling_forward::primitive_desc>
+  AcquireForwardPrimitiveDescriptor() {
```

Comment:
1) Why do you need to define your own AcquireForwardPrimitiveDescriptor. There is generic implementation in MKLDNNHandlerT . Why not use this base one?

2) Currently for primitivedescriptor no info on threads should be added to the key. you used key_ which (for Parallel executor, and dygraph) will contain TID which makes crash during training

Reply:
I called AcquireForwardPrimitiveDescriptor in MKLDNNHandlerT now